### PR TITLE
support hexadecimal input for `--embed`

### PR DESCRIPTION
**Pull request by ssiyad**
https://github.com/alacritty/alacritty/issues/6145

### CODE REVIEW

**Review by kchibisov — CHANGES_REQUESTED** [review]
In general the approach looks good.

However I'd probably write something like

```diff
diff --git a/alacritty/src/cli.rs b/alacritty/src/cli.rs
index be377700..babdcafa 100644
--- a/alacritty/src/cli.rs
+++ b/alacritty/src/cli.rs
@@ -100,7 +100,13 @@ impl Options {
         }
 
         config.window.dynamic_title &= self.window_options.window_identity.title.is_none();
-        config.window.embed = self.embed.as_ref().and_then(|embed| embed.parse().ok());
+        config.window.embed = self.embed.as_ref().and_then(|embed| {
+            embed
+                .starts_with("0x")
+                .then(|| u64::from_str_radix(&embed[2..], 16).ok())
+                .unwrap_or_else(|| embed.parse().ok())
+        });
+
         config.debug.print_events |= self.print_events;
         config.debug.log_level = max(config.debug.log_level, self.log_level());
         config.debug.ref_test |= self.ref_test;
```

And probably put it into `parse_hex_or_decimanl(&str)` function with a test for it. You can look at `parse_class`, so the end result will look like

```rust
config.window.embed = self.embed.as_ref(),and_then(parse_hex_or_decimal)`
```

**Review by kchibisov — CHANGES_REQUESTED** [review]
Looks good, could you add an entry to `Changed` section in `CHANGELOG.md`, like: `--embed option could accept hexadecimal values by providing `0x`.

Also, you should tune `--help` output, since right now it only states `decimal`, but not hexadecimal starting with `0x`.

File: alacritty/src/cli.rs [hunk 1]
```diff
@@ -173,6 +173,14 @@ fn parse_class(input: &str) -> Result<Class, String> {
     }
 }
 
+/// Convert to hex if possible, else decimal
+fn parse_hex_or_decimal(input: &str) -> Option<u64> {
+    input
+        .starts_with("0x")
+        .then(|| u64::from_str_radix(&input[2..], 16).ok())
+        .unwrap_or_else(|| input.parse().ok())
```
Comment (chrisduerr) [inline alacritty/src/cli.rs:181]: I recommend using the `strip_prefix` method here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - The `--help` output was reworked with a new colorful syntax
 - OSC 52 is now disabled on unfocused windows
 - `SpawnNewInstance` no longer inherits initial `--command`
+- `--embed` could accept hexadecimal values starting with `0x`
```
Comment (chrisduerr) [inline CHANGELOG.md:29]: This is a new feature, should be moved to `Added`. "could" is also not appropriate here, I recommend the following:

```
 - Accept hexadecimal values starting with `0x` for `--embed`
```

File: alacritty/src/cli.rs [hunk 3]
```diff
@@ -25,7 +25,7 @@ pub struct Options {
     #[clap(long)]
     pub ref_test: bool,
 
-    /// Defines the X11 window ID (as a decimal integer) to embed Alacritty within.
+    /// Defines the X11 window ID (as a decimal integer or hexadecimal starting with 0x) to embed Alacritty within.
```
Comment (chrisduerr) [inline alacritty/src/cli.rs:28]: I think we can do much better than this.

Maybe something like that:

```
/// X11 window ID to embed Alacritty within (decimal or hexadecimal with "0x" prefix).
```

**Review by chrisduerr — CHANGES_REQUESTED** [review]

File: alacritty/src/cli.rs [hunk 4]
```diff
@@ -173,6 +173,14 @@ fn parse_class(input: &str) -> Result<Class, String> {
     }
 }
 
+/// Convert to hex if possible, else decimal
+fn parse_hex_or_decimal(input: &str) -> Option<u64> {
```
Comment (kchibisov) [inline alacritty/src/cli.rs:177]: Seems like we should use `c_ulong` instead of `u64` here.

**Review by chrisduerr — APPROVED** [review]
LGTM

**Review by kchibisov — APPROVED** [review]

**Comment by ssiyad** [comment]
@chrisduerr @kchibisov As a rust newbie, I couldn't have done this without your guidance. Thank you!

**Pull request merged**